packages/legend-graph/src/models/protocols/pure/v1/engine/execution/V1_ExecuteInput.ts

```diff
@@ -55,3 +56,24 @@ export class V1_ExecuteInput {
     }),
   );
 }
+
+export class V1_TestDataGenerationExecutionInput extends V1_ExecuteInput {
+  parameters: (string | number | boolean)[] = [];
+  hashStrings = false;
+
+  static _serialization = new SerializationFactory(
```

Comment:
I believe we can do `static override serialization` here

Reply:
Nope because of

// NOTE: due to contra-variance and variance, this class will not work for polymorphism
// hence, use `BasicSerializationFactory`


We could leverage  `BasicSerializationFactory`
but that would make redefined toJson, fromJson on every V1_ExecutionInput which is more used. Figured this is a cleaner approach. Maybe a note ?

Reply:
sure, a note should be fine

Reply:
I just tried, looks like we can change `BasicSerializationFactory` to `SerializationFactory` for `V1_ExecuteInput`

Reply:
Thanks to you I discovered some TS quirks I never faced before. If we rewrote `SerializationFactory` to

```ts

export class SerializationFactory<T> extends BasicSerializationFactory<T> {
  toJson(val: T): PlainObject<T> {
    return serialize(this.schema, val);
  }
  fromJson(val: PlainObject<T>): T {
    return deserialize(this.schema, val);
  }
}

```

we're good. Anyway, leave this as is, I'll deal with it